transcom/mymove
# B-20671 B-20674 B-20758 simulate process TPPS cmd INT

## [B-20671](https://github.com/transcom/mymove/pull/13434)
## [B-20674](https://github.com/transcom/mymove/pull/13510)
## [B-20758](https://github.com/transcom/mymove/pull/13554)

> [!IMPORTANT]  
> The commented out code in process_edis.go and gex_sftp.go will be uncommented in B-20560, with gov PO approval. Commenting out that code out will allow us to push this code to staging/prod without seeing errors in the logs while we wait on the final TPPS SFTP directory. Added to Agility [here](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20560) and [here](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20671).

## Summary

This will let us "process" a TPPS report.

Cons:
- The TPPS data file used is directly from TPPS - meaning it contains prod data.
- Since the invoice numbers and all payment data from the prod TPPS report won't match payment_request_numbers from the testing environments, it will require some database manipulation to be able to insert the prod data into the relevant environment's database (step 3 of the how to test section)
- The payment data for each service item displayed on the UI won't match, as it's again prod vs test environment payment information

Pros:
- Let's us prove that the processing of a file works to populate the database, updates the payment request status to PAID, and let's us test frontend changes in the loadtest environment, without waiting on the once daily stg/prod ecs task
- Will be able to view the TPPS Paid status as well as total TPPS paid amounts on the UI (can view service item individual amounts if the test payment request has a matching reservice code to the test file, but amounts will likely differ)

### How to test

1. Checkout the branch and run `make build`
2. In the db, pick out four payment requests with a status of REVIEWED
3. Update their `payment_request_number`s to be 1077-4079-3, 1208-5962-1, 8801-2773-2, and 8801-2773-3
4. Run `go run ./cmd/simulate-process-tpps/`
5. Check the terminal output that tells you what data should now appear in the tpps_paid_invoice_reports table and verify that it matches
6. Verify that for the four payment requests, the status is now PAID, you can use the below script or search by the payment request number to find them in the TIO queue.

```
select m."locator" from payment_requests pr
join moves m on m.id = pr.move_id
where pr.payment_request_number = '1077-4079-3'
or pr.payment_request_number = '1208-5962-1'
or pr.payment_request_number = '8801-2773-2'
or pr.payment_request_number = '8801-2773-3'

```
7. Feel free to check the UI but just know it will be a little wonky for the payment amounts - but you should see the TPPS PAID flag at the very least for each of the four payment requests.
